jdbc/core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/driver/ShardingSphereDriverURL.java

```diff
@@ -57,7 +61,15 @@ public ShardingSphereDriverURL(final String url) {
     @SneakyThrows(IOException.class)
     public byte[] toConfigurationBytes() {
         try (InputStream stream = inClasspath ? ShardingSphereDriverURL.class.getResourceAsStream("/" + file) : Files.newInputStream(new File(file).toPath())) {
-            byte[] result = new byte[null == stream ? 0 : stream.available()];
+            BufferedReader reader = new BufferedReader(new InputStreamReader(stream, Charsets.UTF_8));
+            StringBuilder builder = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (!line.startsWith("#")) {
+                    builder.append(line);
+                }
+            }
+            final byte[] result = builder.toString().getBytes(StandardCharsets.UTF_8);
             Objects.requireNonNull(stream, String.format("Can not find configuration file `%s`.", file)).read(result);
```

Comment:
Seems `Objects.requireNonNull` should be put after `try-resource` and before real reading